### PR TITLE
Cal heatmap controls migration

### PR DESCRIPTION
**Pull request by rusackas**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Moving unique control(s) from `controls.jsx` to `/explore/controlPanels/CalHeatmap.js`

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/incubator-superset/issues/9187
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@kristw

### CODE REVIEW

File: superset-frontend/src/explore/controlPanels/CalHeatmap.js
Comment (rusackas) [inline superset-frontend/src/explore/controlPanels/CalHeatmap.js]: @kristw it appears I can change this one to use `x_axis_format` with a couple config overrides. This would get rid of `x_axis_time_format` entirely, which appears to be a deprecated practice. 

If that's a good idea, I can make the change here, and a quick PR in `superset-ui-plugins` to match. 

However, I'm not sure if this leads to any painful migrations for existing charts/dashboards. Thoughts?

Comment (kristw) [reply superset-frontend/src/explore/controlPanels/CalHeatmap.js]: Umm, as much as I wanna merge them, this will cause `db:migration` for existing charts/dashboards so perhaps let's hold on to it.

**Review by kristw — APPROVED** [review]

**Comment by williaster** [comment]
so excited for this! 👯

File: superset-frontend/src/explore/controlPanels/CalHeatmap.js
Comment (rusackas) [reply superset-frontend/src/explore/controlPanels/CalHeatmap.js]: Ok... I'll try to keep track of that, as it may be worth addressing this and other similar migrations... maybe we can get a handful of them in a batch. Heads up to @villebro or @willbarrett that this may come up in discussion.

Comment (villebro) [reply superset-frontend/src/explore/controlPanels/CalHeatmap.js]: @runsackas I like the idea of doing a migration batch to clean up duplicate controls. Let's hold off on the changes for now and put them in a google sheet or similar so we can do one or two migrations when the control migration is otherwise done.

**Pull request merged**
